force-app/main/default/lwc/refundPayment/refundPayment.js

```diff
@@ -0,0 +1,124 @@
+
+import PAYMENT_AMOUNT_FIELD from '@salesforce/schema/npe01__oppPayment__c.npe01__Payment_Amount__c';
+import PAYMENT_DATE_FIELD from '@salesforce/schema/npe01__oppPayment__c.npe01__Payment_Date__c';
+
+export default class refundPayment extends LightningElement {
+    @api recordId;
+    hasError = false;
+    errorMessage;
+    labels = Object.freeze({
+        refundPaymentTitle,
+        refundPaymentConfirmButton,
+        cancelButtonLabel,
+        commonRefreshPage,
+        noRefundPermissionMessage,
+        refundPaymentMessage,
+        refundProcessing,
+        refundPaymentErrorMessage
+    });
+    refundView;
+    paymentAmount;
+    paymentDate;
+
+    @wire(getRecord, {
+        recordId: '$recordId',
+        fields: [PAYMENT_AMOUNT_FIELD, PAYMENT_DATE_FIELD]})
+    wiredRecord({ error, data }) {
+        if (error) {
+            this.hasError = true;
+            this.errorMessage = constructErrorMessage(error).detail;
+        } else if (data) {
+            this.paymentAmount = getFieldValue(data, PAYMENT_AMOUNT_FIELD);
+            this.paymentDate = getFieldValue(data, PAYMENT_DATE_FIELD);
+        }
+    }
+
+    get paymentDateMessage() {
+        return this.labels.refundPaymentMessage.split(/{[^}]*}/)[0];
+    }
+
+    get paymentAmountMessage() {
+        return this.labels.refundPaymentMessage.split(/{[^}]*}/)[1];
+    }
+
+    async connectedCallback() {
+        try {
+            this.refundView = await getPermissionData();
+            if (this.refundView.hasRequiredPermissions === false) {
+                this.displayErrorMessage(this.labels.noRefundPermissionMessage);
+            }
+
+        }catch (ex) {
+            this.displayErrorMessage(constructErrorMessage(ex).detail);
+        }
+    }
+
+    handleRefund() {
+        processRefund({
+            paymentId: this.recordId
+        }) 
+            .then((response) => {
+                this.processResponse(response);
+        })
+        .catch((error) => {
+            this.displayErrorMessage(constructErrorMessage(error).detail);
+        });
+
+    }
+
+    handleClose(){
+        this.dispatchEvent( new CloseActionScreenEvent() );
+    }
+
+    processResponse(response) {
+        if (response.hasRequiredPermissions === false) {
+            this.displayErrorMessage(this.labels.noRefundPermissionMessage);
+            return;
+
+        } else if (response.isSuccess === true) {
+            this.showToast('', this.labels.refundProcessing + ' {0}', 'info', '', [
+                {
+                    url: '/' + response.redirectToPaymentId,
+                    label: this.labels.commonRefreshPage,
+                }]
+            );
+
+        } else if (response.isSuccess === false) {
+            this.showToast(this.labels.refundPaymentErrorMessage, response.errorMessage, 'error');
+        }
+        
+        this.handleClose();
+    }
+
+    showToast(title, message, variant, mode, messageData){
```

Comment:
I think we have an existing `showToast` function in the `utilCommon` class that we can use here.